### ISSUE TITLE
Unit test case `goodProducer` fails on 32 bit architectures

### ISSUE THREAD

**Issue by nomeata** [bos/aeson#333]
Hi,

the test case added in b024cd20155d3c17b823cac0835f7e3e8677371c fails consistently on non-64-bit architectures, e.g. in https://buildd.debian.org/status/fetch.php?pkg=haskell-aeson&arch=armel&ver=0.10.0.0-2&stamp=1452664317 we have
```
unit:
  camelCase:
    camelTo: [OK]
    camelTo: [OK]
    camelTo: [OK]
    camelTo: [OK]
    camelTo2: [OK]
    camelTo2: [OK]
    camelTo2: [OK]
    camelTo2: [OK]
  encoding:
    goodProducer: [Failed]
ERROR: Prelude.undefined
```

No idea about the cause, sorry, but it is a problem for packaging aeson for Debian.

Thanks,
Joachim

**Comment by nomeata** [comment]
On other architectures, there are more problems, e.g. on `mips`:
```
data families:
  template-haskell:
    toEncoding:
      NullaryString: [OK, passed 100 tests]
      Nullary2ElemArray: [OK, passed 100 tests]
      NullaryTaggedObject: [Failed]
*** Failed! Falsifiable (after 1 test): 
C3
"{\"contents\":[],\"tag\":\"c3\"}" /= "{\"tag\":\"c3\",\"contents\":[]}"
(used seed -1077253334)

properties:
  template-haskell:
    toEncoding:
      NullaryString: [OK, passed 100 tests]
      Nullary2ElemArray: [OK, passed 100 tests]
      NullaryTaggedObject: [Failed]
*** Failed! Falsifiable (after 1 test): 
C2
"{\"contents\":[],\"tag\":\"c2\"}" /= "{\"tag\":\"c2\",\"contents\":[]}"
(used seed -1847255911)
```
https://buildd.debian.org/status/fetch.php?pkg=haskell-aeson&arch=mips&ver=0.10.0.0-2&stamp=1452663040

**Comment by bos** [comment]
I don't have any 32-bit or exotic hardware. I'll likely need your help to track this down.

**Comment by nomeata** [comment]
It should be reproducible in a 32 bit chroot, but of course I’m will to help. Do you have any idea where to look?

**Comment by nomeata** [comment]
Ok, the latter might be a problem with the design of the test case: If the order of the entries is determined by the order of the hash value of the key, and the hash value depends on the architecture, then this is simply not a valid way to compare them. But this does not explain the `undefined` above.

**Comment by bos** [comment]
I think we have two different bugs. Can you please file the `mips` bug as a separate issue, and we'll keep this one confined to the 32-bit problem?

**Comment by nomeata** [comment]
✓, see #340

**Comment by bos** [comment]
The `goodProducer` test is fixed by a35b5570d6d729a38d68e309c3bd92f3138e3833.

**Comment by nomeata** [comment]
✓, fix confirmed
